src/components/ReportActionItem/ReportActionItemImage.tsx

```diff
@@ -41,37 +45,30 @@ type ReportActionItemImageProps = {
  * and optional preview modal as well.
  */
 
-function ReportActionItemImage({thumbnail, image, enablePreviewModal = false, transaction, canEditReceipt = false, isLocalFile = false}: ReportActionItemImageProps) {
+function ReportActionItemImage({
+    thumbnail,
+    isThumbnail,
+    image,
+    enablePreviewModal = false,
+    transaction,
+    canEditReceipt = false,
+    isLocalFile = false,
+    fileExtension,
+}: ReportActionItemImageProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const imageSource = tryResolveUrlFromApiRoot(image ?? '');
     const thumbnailSource = tryResolveUrlFromApiRoot(thumbnail ?? '');
     const isEReceipt = transaction && TransactionUtils.hasEReceipt(transaction);
 
-    let receiptImageComponent: ReactElement;
+    let propsObj: ReceiptImageProps;
 
     if (isEReceipt) {
-        receiptImageComponent = (
-            <View style={[styles.w100, styles.h100]}>
-                <EReceiptThumbnail transactionID={transaction.transactionID} />
-            </View>
-        );
-    } else if (thumbnail && !isLocalFile && !Str.isPDF(imageSource as string)) {
-        receiptImageComponent = (
-            <ThumbnailImage
-                previewSourceURL={thumbnailSource}
-                style={[styles.w100, styles.h100]}
-                isAuthTokenRequired
-                shouldDynamicallyResize={false}
-            />
-        );
+        propsObj = {isEReceipt: true, transactionID: transaction.transactionID};
+    } else if ((thumbnail ?? isThumbnail) && !isLocalFile && !Str.isPDF(imageSource)) {
+        propsObj = thumbnailSource ? {shouldUseThumnailImage: true, source: thumbnailSource} : {isThumbnail: true, fileExtension, transactionID: transaction?.transactionID};
```

Comment:
What is the difference between `thumbnail` and `isThumbnail` here? I don't think the check for `isThumbnail` and the assignment on `propsObj` for not `thumbnailSource` is not necessary. We can leave that to `else` below.